analyzers/src/SonarAnalyzer.CSharp/Extensions/AssignmentExpressionSyntaxExtensions.cs

```diff
@@ -108,17 +120,15 @@ private static bool MapTupleElements(ImmutableArray<AssignmentMapping>.Builder a
             {
                 var leftExpression = leftEnumerator.Current.Expression;
                 var rightExpression = rightEnumerator.Current.Expression;
-                if (TupleExpressionSyntaxWrapper.IsInstance(leftExpression)
-                    && TupleExpressionSyntaxWrapper.IsInstance(rightExpression))
+                switch (HandleTupleNesting(arrayBuilder, leftExpression, rightExpression))
                 {
-                    if (!MapTupleElements(arrayBuilder, (TupleExpressionSyntaxWrapper)leftExpression, (TupleExpressionSyntaxWrapper)rightExpression))
-                    {
+                    case true:  // Nesting was handled.
+                        break;
+                    case false: // Nesting failed.
                         return false;
-                    }
-                }
-                else
-                {
-                    arrayBuilder.Add(new AssignmentMapping(leftExpression, rightExpression));
+                    case null:  // No nesting.
+                        arrayBuilder.Add(new AssignmentMapping(leftExpression, rightExpression));
+                        break;
```

Comment:
I was a bit confused of `break` in the context of the exisitng outer `while`, I'd suggest adding a small inline comments
```suggestion
                        break; // the switch
```